keylime/tpm2.py

```diff
@@ -429,6 +432,8 @@ def __create_ek(self, asym_alg=None):
                 command = "tpm2_getpubek -H 0x81010007 -g {asymalg} -f {ekpubfile} -P {ekpw} -o {opw} -e {epw}".format(**cmdargs)
             elif tools_version == "4.0":
                 command = "tpm2_createek -c - -G {asymalg} -u {ekpubfile} -p {ekpw} -w {opw} -P {epw}".format(**cmdargs)
+            elif tools_version == "4.2":
```

Comment:
I'm wondering if the `-p` EK password option isn't actually doing anything, even in v4.0, which is why it was removed in v4.2 (see: https://github.com/tpm2-software/tpm2-tools/issues/1917).

Reply:
Ah, good good... I didn't see the bug report (thanks) in `tpm2-tools`, but I concur, even from the commit reported on the opened issue, that it looks like the option `-p` does not have any effect even before 4.2. My main goal on this very first PR of mine for this project was to simply make the minimal amount of changes to get it back to functionality (and I don't have proper facilities for full, multi-version testing, was gonna ask you guys about it actually). As I mentioned before, I would be willing to attempt a second refactoring of it to have something like a "tpm2-tool vs version command map" (perhaps a python dictionary), but would argue in favor of making it on a separate PR. Would you agree with that?

Reply:
@maugustosilva I did consider that before, so I concur a version map would be a sound approach. It would save us building a long list of conditionals each time a new set of tools come out.

Reply:
Just to confirm: you are ok with that (replacing the long list of conditionals) being done on a separate PR, correct?

Reply:
Yeah I agree that sounds reasonable, plus I think we need to look into the consequences of that password option not actually doing anything (and how to fix it).